lib/matplotlib/animation.py

```diff
@@ -859,7 +859,11 @@ def _blit_clear(self, artists, bg_cache):
         # cache and restore.
         axes = set(a.axes for a in artists)
         for a in axes:
-            a.figure.canvas.restore_region(bg_cache[a])
+            try:
+                a.figure.canvas.restore_region(bg_cache[a])
+            except KeyError:
+                # something has happened out of order?
```

Comment:
Do we really want to be swallowing this exception silently? Should we at least attach one of our logging stuff to it?